### DISCUSSION TITLE
Enable `eslint-plugin-mdx` for linting `.mdx` files and a true showcase with `ignoreRemarkConfig` option enabled so `remark-cli` still will be used

### DISCUSSION THREAD

**Discussion by JounQin** [mdx-js/mdx#2425] [General]
`ignoreRemarkConfig` must be enabled because `remark-lint` is not compatible with mdx v3 yet

**Comment by ChristianMurphy** [comment]
Hey @JounQin! 👋 
Are you announcing you are working on a show case?
Or are you asking a question about `remark-lint`?
I'm not sure I follow either the announcement or the question.

**Reply by JounQin** [reply]
@ChristianMurphy Sorry for the confusion.

> Are you announcing you are working on a show case?

Yes, I mean we should use `eslint-plugin-mdx` in `mdx-js/mdx` repository as a showcase. (I was opening this discussion from `mdx-js/mdx`, maybe I should use issue actually?)

**Reply by JounQin** [reply]
@ChristianMurphy Is this change accepted?

**Reply by ChristianMurphy** [reply]
I'm not entirely sure what you are proposing still.

* Downgrading the mdx version in the playground? I'd prefer it used latest.
* Add remark lint and eslint to the playground? Maybe, I'd be cautious about how we visualize and communicate it. The playground is intended as a friendly place to try stuff out. Error and warning messages aren't necessary a friendly intro 😉
* something else?

**Reply by JounQin** [reply]
I mean add `eslint-plugin-mdx` to `mdx-js/mdx` to lint its `.mdx` documents.

**Reply by ChristianMurphy** [reply]
There are a few sets of MDX documents.
Do you mean:
* The documents in the `/docs` folder? (there may be value in linting this more)
* The fixtures in the unit tests? (these intentionally include invalid and confusing documents to ensure the implementation is robust)
* Something else?

I do have some reservations about linting mdx 3 documents with an mdx 2 parser.
Both from:
* a dependency tree perspective, having two copies of every dependency of MDX in the tree could cause weird transitive behavior and could introduce conflicts in TypeScript.
* a parser perspective, v3 has syntax improvements that don't exist in v2, I don't want the documents to be limited to only v2 syntax.

**Reply by JounQin** [reply]
@ChristianMurphy

First of all, `eslint-plugin-mdx` v3 has been released with `mdx` v3 support a long time ago. 🤣

https://github.com/mdx-js/eslint-mdx/releases/tag/v3.0.0

> Rhe documents in the /docs folder? (there may be value in linting this more)

Yes

> The fixtures in the unit tests? (these intentionally include invalid and confusing documents to ensure the implementation is robust)

The fixtures should just be ignored.

**Reply by ChristianMurphy** [reply]
> mdx v3 support a long time ago

to quote your initial comment

> is not compatible with mdx v3 yet

😉
I'm glad to hear there is at least partial support 🙂 


> > The documents in the /docs folder? (there may be value in linting this more)
>
> Yes
>
> > The fixtures in the unit tests? (these intentionally include invalid and confusing documents to ensure the implementation is robust)
>
> The fixtures should just be ignored.

Sounds good to me

**Reply by JounQin** [reply]
@ChristianMurphy 

> is not compatible with mdx v3 yet

I mean `remark-lint`, not `eslint-mdx`.

**Reply by wooorm** [reply]
remark-lint is working in the MDX repo: https://github.com/mdx-js/mdx/blob/d1fface103b90d2ef8d14d50b77cbfcf20b4bb71/package.json#L151.

Personally, I am not sure what the benefits of `eslint-plugin-mdx` would be for `mdx-js/mdx`.
We already use `remark-lint` in the repo, so what would it add?
And we don’t use eslint directly, but use `xo` there.

The repo is also already quite big. Historically the repo was slow to build and check. I made it faster. Adding more tools will make it slower.

**Reply by JounQin** [reply]
That's why `remark-lint` can be disabled when using with `eslint-mdx`, they aim different syntaxes, for example react related jsx rules.

**Reply by wooorm** [reply]
Fine but that doesn’t answer my questions.

**Reply by JounQin** [reply]
Sorry, I don't quite sure which part? The benefit is `eslint-mdx` can help linting jsx rules for mdx documents, and xo should support adding custom eslint plugins?

For faster or slower, I believe linting is more important. Otherwise we should just drop all linting tools?

**Reply by wooorm** [reply]
You can have no food. You can have too much food. Between that there is *enough* food.

I think `xo` and `prettier` and `remark`/`remark-lint` are useful in the 500+ project I work on.
There are other linters that could be used. To use them, I need good reasons.

Do you have examples of things that are bad now? Things that `eslint-mdx` would solve?

**Reply by JounQin** [reply]
The bad part is the ES syntaxes in mdx files are not linted at all? It **could** have some issues, but it's not **must** be there. Some issues maybe already be resolved via human testing/reporting or PR reviews, while `eslint-plugin-mdx` makes them lintable as tools. That's why `eslint-mdx` is built for?

**Reply by wooorm** [reply]
There is almost no JavaScript in the MDX files in `mdx-js/mdx`. The JavaScript that they contain doesn’t change often.

Do you have examples of things that are bad now, that `eslint-mdx` would solve?

**Reply by JounQin** [reply]
Let me try to raise a PR with `eslint-plugin-mdx` and `eslint-plugin-react` enabled to confirm.
